src/lib.rs

```diff
@@ -1,3 +1,4 @@
+#![feature(doc_cfg)]
```

Comment:
I am not comfortable with putting unstable features into this code, because it breaks stable builds (not to mention the MRSV builds). I will comment on the PR about this more fully after I finish this review.

Reply:
Have a quick check at tokio, this is how to make it only confined in the docrs build, https://github.com/tokio-rs/tokio/blob/20c1fdc678a7b9947b1d12d53619c90291f995d2/tokio-util/src/lib.rs#L12

```
#![cfg_attr(docsrs, feature(doc_cfg))]
```